### PR TITLE
added alternative binary install path options

### LINKED ISSUE #1234: Wrangler is not installed properly in gitlab ci through npm
#### Environment

* operating system: gitlab ci node:12
* output of `rustc -V`:
* output of `node -v`:v12.16.3
* output of `wrangler -V`:
* contents of `wrangler.toml`

#### Steps to reproduce
- Create a package.json with @cloudflare/wrangler as a dep
Create an .gitlab-ci.yml with the following content
```
image: node:12

# This folder is cached between builds
# http://docs.gitlab.com/ce/ci/yaml/README.html#cache
cache:
  paths:
    - node_modules/
    - .yarn

before_script:
  - yarn config set cache-folder .yarn

stages:
  - build
  - deploy

build:
  stage: build
  script:
    - yarn
    - wrangler
```

#### What did you expect to see?
- The build should work correctly

#### What did you see instead?
```
 /node_modules/binary-install/src/binary.js:60
       throw `You have not installed ${this.name ? this.name : "this package"}`;
       ^
You have not installed wrangler
 (Use `node --trace-uncaught ...` to show where the exception was thrown)
 Task completed in 0m0.096s
Running after_script
00:01
Uploading artifacts for failed job
00:02
 ERROR: Job failed: exit code 1
```

### LINKED ISSUE #1286: Install Wrangler in local node_modules by default
#### Overview and problem statement

Current install directory is `join(os.homedir(), ".wrangler");`.

`homedir()` is blocked by many CI/CD tools (I tried on Vercel and Gatsby Cloud) which means you can't install anything there. Usually, if you run `npm i <package>` it installs in local `node_modules` - you can use different versions of packages safely in different projects. A package should install globally if you run `npm i -g <package>` command.

### PR DESCRIPTION
**Pull request by bpofficial**
Added environment variables 'WRANGLER_USE_CWD' and 'WRANGLER_INSTALL_PATH' to allow the user to use custom paths to install the wrangler binary. Fixes #1286 and fixes #1234 and allows CI/CD users to correctly publish to cloudflare workers.

### CODE REVIEW

**Comment by bpofficial** [comment]
bumping this

**Review by nataliescottdavidson — APPROVED** [review]
This looks fine to me.

**Comment by nataliescottdavidson** [comment]
@xtuc can you take a look?

File: npm/binary.js [hunk 1]
```diff
@@ -20,17 +20,22 @@ const getPlatform = () => {
 };
 
 const getBinaryURL = (version, platform) => {
-  const site = process.env.WRANGLER_BINARY_HOST ||
-      process.env.npm_config_wrangler_binary_host ||
-      'https://workers.cloudflare.com/get-npm-wrangler-binary';
+  const site =
+    process.env.WRANGLER_BINARY_HOST ||
+    process.env.npm_config_wrangler_binary_host ||
+    "https://workers.cloudflare.com/get-npm-wrangler-binary";
```
Comment (xtuc) [inline npm/binary.js:26]: is there an issue for adding a custom host?

**Comment by xtuc** [comment]
Any way we can test this change?

File: npm/binary.js (hunk 1 above)
Comment (bpofficial) [reply npm/binary.js:26]: #1286

Comment (xtuc) [reply npm/binary.js:26]: but that issue is about wrangler's installation path not changing the endpoint to get the executable

**Review by xtuc — CHANGES_REQUESTED** [review]
We are planning to configure where you can download the wrangler binary from

**Comment by osdiab** [comment]
any timeline when that might happen @xtuc ?

**Comment by xtuc** [comment]
My apologies, I misread the diff and remove my previous objection.

**Review by xtuc — APPROVED** [review]

**Review by ags799 — APPROVED** [review]

**Comment by bpofficial** [comment]
bumping

**Comment by xortive** [comment]
@bpofficial thanks for the contribution!

**Pull request merged**